### PR TITLE
fix: api call relocation

### LINKED ISSUE #255: API call relocation
**Describe the bug**
Some Apis are still called inside components/pages. Those are to be moved to their respective services.

### PR DESCRIPTION
**Pull request by jasirneoito**
## Description

Some Apis are still called inside components/pages. Those are to be moved to their respective services.changed login signup [chapterName] 
 
Fixes #255 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

via postman

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pages/login.js [hunk 1]
```diff
@@ -91,34 +89,16 @@ export default function Login({ referer }) {
       draggable: true,
       progress: undefined
     });
-  const handleClick = (e) => {
+  async function handleClick (e) {
     e.preventDefault();
     setIsLoading(true);
     // console.log(e.target);
     if (validEmail && validPassword) {
       const password = document.querySelector("#password").value;
```
Comment (thealpha93) [inline pages/login.js:97]: you don't need to use `document.queryselector` here.

File: pages/login.js [hunk 2]
```diff
@@ -91,34 +89,16 @@ export default function Login({ referer }) {
...
-          .then((response) => {
-            setIsLoading(false);
-            return response.data;
-          })
-          .then((data) => {
-            // Getting token from cookie
+        const data = await signIn(email, password)
             const cookies = new Cookies();
             const userToken = cookies.get("token");
 						const newDate = new Date(moment().add(30, 'days')).toUTCString();
```
Comment (thealpha93) [inline pages/login.js:104]: code intendation

File: pages/curriculum/[courseName]/[chapterName].js [hunk 3]
```diff
@@ -54,23 +55,12 @@ export default function Chapter({ chapterData, chapterName, courseName }) {
...
-        axios({
-          method: "post",
-          mode: "no-cors",
-          url: `${baseUrl}/api/enrol/${courseName}/${chapterName}`,
-          headers: {
-            "x-access-token": `${cook.accessToken}`
-          }
-        }).then((response) => {
-          if (response.data.entryAdded) {
+          const data = await enrollCourse(courseName, chapterName)
```
Comment (thealpha93) [inline pages/curriculum/[courseName]/[chapterName].js:58]: need `try catch` block

File: pages/signup.js [hunk 4]
```diff
@@ -165,27 +163,9 @@ export default function SignUp({referer}) {
...
-          credentials: 'include',
-          body: JSON.stringify(signupData)
-        })
-          .then((response) => {
-            setIsLoading(false);
-            return response.json();
-          })
-          .then((data) => {
-            // Getting token from cookie
+          const data = await signUp(email, password,fName,username)
```
Comment (thealpha93) [inline pages/signup.js:166]: need `try catch` block

**Review by thealpha93 — CHANGES_REQUESTED** [review]
Use `try catch` block wherever you use await

File: config/config.js [hunk 5]
```diff
@@ -10,6 +10,8 @@ const postsUrl = "api/v1/posts";
 const allPostsUrl = "api/posts";
 const changeStatusUrl = "api/user/post";
 const subscribeUrl = "api/subscribe";
+const loginUrl = "api/v1/signin";
```
[the remark this answers is not in the document]
Comment (jasirneoito) [reply config/config.js:13]: okay

**Pull request merged**